tonic/src/codec/mod.rs

```diff
@@ -63,6 +98,9 @@ pub trait Encoder {
 
     /// Encodes a message into the provided buffer.
     fn encode(&mut self, item: Self::Item, dst: &mut EncodeBuf<'_>) -> Result<(), Self::Error>;
+
+    /// Controls how tonic creates and expands encode buffers.
+    fn buffer_settings(&self) -> BufferSettings;
```

Comment:
to avoid another breaking change here, can we have this be a default fn that just calls `BufferSettings::default()`?

Reply:
and the same for the other trait.

Reply:
That seems very sensible, thank you